### DISCUSSION TITLE
transports.file.archiveLog not working on folder different than /%appData%/AppName/logs

### DISCUSSION THREAD

**Discussion by idaamit** [megahertz/electron-log#326] [Q&A]
I would like to use  `log.transports.file.archiveLog ` but to keep the log location as it was in version 2.2.1 
I have changed the location via log.transports.file.resolvePath
but   `log.transports.file.archiveLog ` is not triggered, only once at the startup.

Any ideas how can I keep the old log location and have  `log.transports.file.archiveLog ` working  ?

**Comment by megahertz** [comment]
`archiveLog` is called once file size exceeds the limit, not at the startup. If it doesn't work correctly, you could override `archiveLog` to check why that doesn't work correctly in your case.

**Reply by idaamit** [reply]
I would like to make situation clearer.
If at start up the log size already exceeds 1M,  `log.transports.file.archiveLog` is being called as expected, and log file is emptied as expected.
The program continues to write to the log, and after some time the log size, exceeds 1M again  (I waited till log size 1.2M), but  `log.transports.file.archiveLog` is not triggered. 
The only configuration that I made was : 

```
log.transports.file.resolvePath = variables => {
    return path.join(variables.appData, variables.appName, `ScreenAgent.log`);
};
```


If I add  the following code, the behaviour is not changed : 
```
log.transports.file.archiveLog = (file) => {
    file = file.toString();
    const info = path.parse(file);

    try {
        fs.renameSync(file, path.join(info.dir, info.name + `.old.ida-${new Date().getTime()}` + info.ext));
    } catch (e) {
        console.warn('Could not rotate log', e);
    }
}
```

**Reply by megahertz** [reply]
In most cases it works fine, but it may calculate file size incorrectly in a case when both main and renderer processes write to the same file. Anyway, I'm working on v5 where the behavior will be changed.
